stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -316,6 +357,13 @@ public void onClose(Status status, Metadata trailers) {
         observer.onError(status.asRuntimeException(trailers));
       }
     }
+
+    @Override
+    public void onReady() {
+      if (adapter.onReadyHandler != null) {
+        adapter.onReadyHandler.run();
```

Comment:
In the case of clients we expect the onReady handler to be bound before call start using the StreamObservers.capture method so there should be no raciness here. I have some tidy-ups to do with that